Traceback (most recent call last):
  File "/home/guglielmo/.local/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/guglielmo/.local/lib/python3.8/site-packages/nbclient/client.py", line 1117, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/guglielmo/.local/lib/python3.8/site-packages/nbclient/util.py", line 78, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/guglielmo/.local/lib/python3.8/site-packages/nbclient/util.py", line 57, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/guglielmo/.local/lib/python3.8/site-packages/nbclient/client.py", line 558, in async_execute
    await self.async_execute_cell(
  File "/home/guglielmo/.local/lib/python3.8/site-packages/nbclient/client.py", line 862, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/home/guglielmo/.local/lib/python3.8/site-packages/nbclient/client.py", line 765, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from scipy import signal
import matplotlib.pyplot as plt
import numpy as np

T=1/1024 # Sampling time
omega = np.arange(start=np.log10(5), stop=np.log1o(0.91*np.pi/T), step=T)
f3dB = 10 # Hz

omega0=2*np.pi*f3dB
tau = 1/(2*np.pi*f3dB)
##print("T=\t{:e}\ntau=\t{:e}".format(T, tau))
fig, ax = plt.subplots(1, 2, figsize=(15, 5))


lp = 1/(1-1j*omega*tau) # Ideal Low pass 
mag_lp = 20*np.log10(np.abs(lp))
phase_lp = np.arctan(np.imag(lp)/np.real(lp))*180/np.pi


c = (1-T/(2*tau) ) / (1+T/(2*tau) )
lp_bt = 0.5*(1-c)*(np.exp(-1j*omega*T)+1 )/( c*np.exp(-1j*omega*T)-1)
mag_lp_bt = 20*np.log10(np.abs(lp_bt))
phase_lp_bt = -1*np.arctan( np.imag(lp_bt)/np.real(lp_bt))*180/np.pi


ax[0].semilogx(omega, mag_lp, label="LP")    # Bode magnitude plot
ax[0].semilogx(omega, mag_lp_bt, "--", label="Bilinear")    # Bode magnitude plot

ax[1].semilogx(omega, phase_lp, label="LP")
ax[1].semilogx(omega, phase_lp_bt, "--", label="Bilinear")

ax[0].vlines(omega0, min(mag_lp_bt), 20*np.log10(np.abs(1/(1-1j))),linestyles=":", label=r"$2\pi f_{3dB}$")
ax[1].vlines(omega0, 0, 45, linestyles=":", label=r"$2\pi f_{3dB}$ [$\simeq$45Â°]")


ax[0].set_title("Magnitude");  ax[1].set_title("Phase")
ax[0].set_xlabel(r"$\omega$ [Hz]"); ax[0].set_ylabel(r"Magnitude [dB]");
ax[1].set_xlabel(r"$\omega$ [Hz]"); ax[1].set_ylabel(r"Phase [degÂ°]");
ax[0].grid();   ax[1].grid()
ax[0].legend(); ax[1].legend()

fig.tight_layout()
plt.show()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
[0;32m/tmp/ipykernel_16994/2228569199.py[0m in [0;36m<module>[0;34m[0m
[1;32m      4[0m [0;34m[0m[0m
[1;32m      5[0m [0mT[0m[0;34m=[0m[0;36m1[0m[0;34m/[0m[0;36m1024[0m [0;31m# Sampling time[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 6[0;31m [0momega[0m [0;34m=[0m [0mnp[0m[0;34m.[0m[0marange[0m[0;34m([0m[0mstart[0m[0;34m=[0m[0mnp[0m[0;34m.[0m[0mlog10[0m[0;34m([0m[0;36m5[0m[0;34m)[0m[0;34m,[0m [0mstop[0m[0;34m=[0m[0mnp[0m[0;34m.[0m[0mlog1o[0m[0;34m([0m[0;36m0.91[0m[0;34m*[0m[0mnp[0m[0;34m.[0m[0mpi[0m[0;34m/[0m[0mT[0m[0;34m)[0m[0;34m,[0m [0mstep[0m[0;34m=[0m[0mT[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      7[0m [0mf3dB[0m [0;34m=[0m [0;36m10[0m [0;31m# Hz[0m[0;34m[0m[0;34m[0m[0m
[1;32m      8[0m [0;34m[0m[0m

[0;32m/usr/local/lib/python3.8/dist-packages/numpy/__init__.py[0m in [0;36m__getattr__[0;34m(attr)[0m
[1;32m    311[0m                 [0;32mreturn[0m [0mTester[0m[0;34m[0m[0;34m[0m[0m
[1;32m    312[0m [0;34m[0m[0m
[0;32m--> 313[0;31m             raise AttributeError("module {!r} has no attribute "
[0m[1;32m    314[0m                                  "{!r}".format(__name__, attr))
[1;32m    315[0m [0;34m[0m[0m

[0;31mAttributeError[0m: module 'numpy' has no attribute 'log1o'
AttributeError: module 'numpy' has no attribute 'log1o'

